### PR TITLE
Feature/dashboard toggle goal topic all

### PR DESCRIPTION
**Pull request by KyleRessLiere**
Added toggle content type to the dashboard. Toggles goals topics and a all view

image
image
image

### CODE REVIEW

File: client/agora/views/dashboard/partials/gallery/note-gallery.ejs [hunk 1]
```diff
@@ -82,43 +83,100 @@
...
+                     } %>
                       <% }}%>
               </div>
+            
 
 
 
              
 
                 <script>
```
Comment (briangormanly) [inline client/agora/views/dashboard/partials/gallery/note-gallery.ejs:134]: No inline scripts!!! :)
all JavaScript MUST be in .js file!

File: client/agora/views/dashboard/partials/gallery/note-gallery.ejs [hunk 2]
```diff
@@ -82,43 +83,100 @@
...
+                  const cards = document.getElementsByClassName("col");
+                  console.log(cards[0])
+                let filter = input.value
+                for (let i = 0; i < cards.length; i++) {
+                let title = cards[i].querySelector(".card-body");
+                    cards[i].classList.remove("d-none")
+                
+            }
+        }
                   $('.dropdown-content').on('click', function (ev) {
```
Comment (briangormanly) [inline client/agora/views/dashboard/partials/gallery/note-gallery.ejs:180]: Is this jquery?? I would prefer to keep the JS vanilla! 
There is no reason to use a library when you can simply write:
document.getElementbyClassName('dropdown-content').addEventListener('click', (ev) => { ... });

**Review by briangormanly — CHANGES_REQUESTED** [review]
I made some notes in the code review regarding the JS use.. 
Sorry if I missed some of this in a previous PR
1. We want to keep all JavaScript in .js files, no <script> blocks in ejs!
2. There is no need anymore for libraries like JQuery, it just makes the code base feel fragmented and confusing. There is no reason to write something differently in JQuery that can be written in plain JS. And there is nothing that you can write in JQuery that cannot be written in modern JS.

File: client/agora/views/dashboard/partials/gallery/note-gallery.ejs (hunk 1 above)
Comment (KyleRessLiere) [reply client/agora/views/dashboard/partials/gallery/note-gallery.ejs]: Ok sounds good just resolved that. I am creating a task for changing jquery to js so the person who wrote code can covert what the have done. However the subject of the pull request has been adjusted to be in the dashboard.js file.

**Review by briangormanly — APPROVED** [review]
Okay looks good.

**Review by sbrad93 — APPROVED** [review]

**Pull request merged**